### DISCUSSION TITLE
Why are there no scrollbars?

### DISCUSSION THREAD

**Discussion by Mandrake-Lee** [Igalia/cog#309] [Q&A]
Hi,

I'm testing cog directly as shell DRM and under Weston/FDO.

In both scenarios, the're no scrollbars that allow to reach all the rendered page.

Therefore, most of the content is lost.

Is there a flag to be activated somewhere?

Thanks

**Comment by aperezdc** [comment]
Hello @Mandrake-Lee! You need to use [WPE WebKit 2.30.0](https://wpewebkit.org/release/wpewebkit-2.30.0.html) or newer for scrollbars to be rendered. That was the first version where support for scrollbars was present. Note that the scrollbars are “overlay” style, which means you will need to move the mouse pointer to the edge to completely show them.

Another possibility is defining the appearance of scrollbars using CSS with [::-webkit-scrollbar](https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-scrollbar). You can either add the needed CSS rules in your web content, or as an [injected CSS style sheet](https://wpewebkit.org/reference/wpewebkit/unstable/WebKitUserContentManager.html#webkit-user-content-manager-add-style-sheet). This should work for versions of WPE WebKit older than 2.30.0 as well.

I hope this helps 😃

**Reply by Mandrake-Lee** [reply]
> 
> 
> Hello @Mandrake-Lee! You need to use [WPE WebKit 2.30.0](https://wpewebkit.org/release/wpewebkit-2.30.0.html) or newer for scrollbars to be rendered. That was the first version where support for scrollbars was present. Note that the scrollbars are “overlay” style, which means you will need to move the mouse pointer to the edge to completely show them.
> 
> Another possibility is defining the appearance of scrollbars using CSS with [::-webkit-scrollbar](https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-scrollbar). You can either add the needed CSS rules in your web content, or as an [injected CSS style sheet](https://wpewebkit.org/reference/wpewebkit/unstable/WebKitUserContentManager.html#webkit-user-content-manager-add-style-sheet). This should work for versions of WPE WebKit older than 2.30.0 as well.
> 
> I hope this helps smiley

Hello,

I've been checking this and now I can see the scrollbars behind weston (FDO platform)

My issue was that the cog canvas was not fitting into the weston screen and therefore the scrollbars were not accessible/visible.

Adjusting the canvas via:
```
COG_PLATFORM_FDO_VIEW_FULLSCREEN=1 cog --platform=fdo
```
or more customized to my needs:
```
COG_PLATFORM_FDO_VIEW_WIDTH=800 COG_PLATFORM_FDO_VIEW_HEIGHT=460 cog --platform=fdo
```
will do the trick.

Note: I cannot check DRM since the mouse is not recognized. This is another issue.

In parallel, @aperezdc suggested to inject CSS style sheet. I'm very interested into this option since the scrollbars are very narrow for my screen. Also I'd like to test is we can inject a javascript keyboard aiming DRM platform. This could be a workaround.

Can injection be done via userspace? Could you provide a howto link?

Thanks in advance!
